### PR TITLE
feat: handle password change return value HP-2501

### PR DESCRIPTION
**Pull request by Riippi**
- Handle return value when changing the password
- Show success notification
- Fix styling on mobile screen size

### CODE REVIEW

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr376-ui.dev.hel.ninja** :rocket::rocket::rocket:

File: src/passwordChange/PasswordChangeCallback.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  history,
+}: RouteChildrenProps): React.ReactElement | null {
+  const { t } = useTranslation();
+  const redirectToErrorPage = useErrorPageRedirect();
+
+  const { setPasswordUpdateState } = useContext(ProfileContext);
+
+  useEffect(() => {
+    const params = new URLSearchParams(document.location.search);
+    const stat = params.get('kc_action_status');
```
Comment (mikkojamG) [inline src/passwordChange/PasswordChangeCallback.tsx:20]: Would this work with react-router-dom hooks, useLocation and/or useParams? 🤔

File: src/profile/components/editingNotifications/EditingNotifications.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { EditDataType } from '../../helpers/editData';
 
 type Props = {
   content: NotificationContent;
-  dataType: EditDataType;
+  dataType: EditDataType | 'password';
```
Comment (mikkojamG) [inline src/profile/components/editingNotifications/EditingNotifications.tsx:11]: Can this union type be added directly to `EditDataType ` ? 🤔

**Review by mikkojamG — COMMENTED** [review]
Looks very good 👍  Had couple questions:

File: src/profile/components/editingNotifications/EditingNotifications.tsx (hunk 2 above)
Comment (Riippi) [reply src/profile/components/editingNotifications/EditingNotifications.tsx:11]: EditDataType is used in so many other places that I was afraid of messing with it. Password change is quite different and uses only this notification.

File: src/passwordChange/PasswordChangeCallback.tsx (hunk 1 above)
Comment (Riippi) [reply src/passwordChange/PasswordChangeCallback.tsx:20]: Switched this to useLocation.

**Review by mikkojamG — APPROVED** [review]

**Pull request merged**